controllers/volsync/vshandler.go

```diff
@@ -1571,3 +1616,295 @@ func ValidateObjectExists(ctx context.Context, c client.Client, obj client.Objec
 
 	return nil
 }
+
+func (v *VSHandler) reconcileLocalReplication(rdSpec ramendrv1alpha1.VolSyncReplicationDestinationSpec,
+	pskSecretName string, l logr.Logger) (*volsyncv1alpha1.ReplicationDestination,
+	*volsyncv1alpha1.ReplicationSource, error,
+) {
+	lrd, err := v.reconcileLocalRD(rdSpec, pskSecretName)
+	if lrd == nil || err != nil {
+		return nil, nil, err
+	}
+
+	rsSpec := &ramendrv1alpha1.VolSyncReplicationSourceSpec{
+		ProtectedPVC: rdSpec.ProtectedPVC,
+	}
+
+	_, lrs, err := v.reconcileLocalRS(rsSpec, pskSecretName, *lrd.Status.RsyncTLS.Address, false)
+	if err != nil {
+		return lrd, nil, err
```

Comment:
I suggest that you always return nil, nil, err in the error case.

Otherwise there is potential problem in ReconcileRD() as it has the following condition
```
if lrd == nil || lrs == nil || err != nil {
			return nil, err
		}
```

and may return a nil, nil unexpectedly.